site/content/features/stringtemplates.md

```diff
@@ -0,0 +1,466 @@
+}
+...
+String result = box("{", "}")."Hello, \{name}! Next year, you'll be \{age + 1}.";
+```
+
+String processors can protect against injection attacks. When creating HTML, XML, JSON, SQL, and so on, from arbitrary values, you want to escape special characters such as quotation marks and bracket delimiters. The JEP has a useful example of using a `PreparedStatement` for a SQL query. This sandbox shows a simpler example for processing XML.
+
+{{< sandbox version=java19 preview="true" mainclass="Xml" >}}{{< sandboxsource "Xml.java" >}}
+import java.io.*;
+import javax.xml.parsers.*;
+import org.w3c.dom.*;
+import org.w3c.dom.ls.*;
+import org.xml.sax.*;
+
+public class Xml {
+   private static String escape(String str) {
+      return str.replace("&", "&amp;")
+         .replace("<", "&lt;")
+         .replace(">", "&gt;")
+         .replace("\"", "&quot;")
+         .replace("'", "&apos;");
+   }
+
+   public static StringTemplate.Processor<Document, Exception> XML = 
+      (StringTemplate template) -> {
+         String escaped = StringTemplate.interpolate(template.fragments(),
+            template.values().stream().map(String::valueOf).map(Xml::escape).toList());
+         return parse(escaped);
+   };
+
+   // Here are two methods for parsing and printing XML. Don't pay too much attention
+   // to the byzantine API from an earlier age...
+   private static Document parse(String xml) throws SAXException, IOException, ParserConfigurationException {
+      DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder builder = factory.newDocumentBuilder();
+      return builder.parse(new ByteArrayInputStream(xml.getBytes()));
+   }
+
+   private static String prettyPrint(Document doc) {
+      DOMImplementation impl = doc.getImplementation();
+      var implLS = (DOMImplementationLS) impl.getFeature("LS", "3.0");
+      LSSerializer ser = implLS.createLSSerializer();
+      ser.getDomConfig().setParameter("format-pretty-print", true);
+      return ser.writeToString(doc);
+   }
+   
+   public static void main(String[] args) throws Exception {
+      String name = "J.R. \"Bob\" Dobbs <dobbs@subgenius.com>";
+      int age = 42;
+      Document doc = XML."""
+<div>
+   <p class="greeting">Hello, \{name}!</p>
+   <p>Next year, you'll be \{age + 1}.</p>
+</div>""";
+      System.out.println(prettyPrint(doc));
+   }
+}
+
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+Now it's your turn. Complete the code below so that one can write a `Map<String, Object>` as 
+
+```
+MAP."key1 \{value1} key2 \{value2} ..."
+```
+
+Trim the fragments and ignore the last one.
+
+{{< sandbox version=java19 preview="true" mainclass="Challenge" >}}{{< sandboxsource "Challenge.java" >}}
+import java.time.*;
+
+public class Challenge {
+   public static ... MAP = ...;
+   public static void main(String[] args) {
+      LocalDate today = LocalDate.now();
+      System.out.println(MAP."""
```

Comment:
I was a bit confused about the objective of this interface. Probably showing the expected type here would have helped me:

    Map<String, Object> map = Map."""
    ...
    System.out.println(map);